user-interface/0001-WALL-wallet.md

```diff
@@ -0,0 +1,186 @@
+  - `TODO` tainted keys **must** not be selectable (<a name="0001-WALL-073" href="#0001-WALL-073">0001-WALL-073</a>)**
+- **must** enter wallet passphrase before wallet details are shared (assuming a password has not recently been entered)(<a name="0001-WALL-022" href="#0001-WALL-022">0001-WALL-022</a>)
+- **must** be able to retrospectively revoke Dapp's access to a Wallet (<a name="0001-WALL-023" href="#0001-WALL-023">0001-WALL-023</a>)
+- **must** be able to retrospectively revoke Dapp's access particular keys in a wallet (<a name="0001-WALL-072" href="#0001-WALL-072">0001-WALL-072</a>)
+
+... so that I can control what public keys are shared with a dapp and what dapps can prompt me to sign transactions 
+
+## Approving transactions
+When a dapp sends a transaction to the wallet for signing and broadcast, I...
+
+- **must** be prompted to approve, reject or ignore the transaction (if auto approve is not on) (<a name="0001-WALL-024" href="#0001-WALL-024">0001-WALL-024</a>)
+- **must** see the details of the transaction. See [details of transaction](#transaction-detail). (<a name="0001-WALL-025" href="#0001-WALL-025">0001-WALL-025</a>)
+- **must** see any ignored/dismissed in a transactions area along with pending, approved and rejected transactions (e.g. history). See [Transactions](#transactions).
+
+... so I can verify that the transaction being sent is the one I want
+
+## View keys balances + positions/accounts
+When looking for a specific balance or asset on a given wallet and network, I...
+
+- **must** see total balances of an asset for each key (<a name="0001-WALL-027" href="#0001-WALL-027">0001-WALL-027</a>)
+- **should** see a breakdown of all accounts for an asset on a key (<a name="0001-WALL-028" href="#0001-WALL-028">0001-WALL-028</a>)
+- **should** see a summary of balances when switching keys (to help me find the key I am looking for) (<a name="0001-WALL-029" href="#0001-WALL-029">0001-WALL-029</a>)
+- **would like to** be able to search by market name/code, and see all keys with a margin or liquidity account in matching markets (<a name="0001-WALL-030" href="#0001-WALL-030">0001-WALL-030</a>)
+- **should** be able to search by asset name or code, and see all keys with balance of matching assets (<a name="0001-WALL-031" href="#0001-WALL-031">0001-WALL-031</a>)
+- **should** be able to search for arbitrary metadata added by user to keys (<a name="0001-WALL-032" href="#0001-WALL-032">0001-WALL-032</a>)
+- **must** be able to see a total of all asset for all keys in a given wallet (<a name="0001-WALL-033" href="#0001-WALL-033">0001-WALL-033</a>)
+ 
+... so that I can find the keys that I am looking for, see how much I have to consolidate (via transfers) or withdraw funds
+
+## Transactions
+When thinking about a recent or specific transaction, I ...
+
+- **must** see a history of transactions the wallet app has signed read from the local app (<a name="0001-WALL-034" href="#0001-WALL-034">0001-WALL-034</a>)
```

Comment:
Might be worth clarifying that this is only for the current session (until app is quit)

Reply:
@barnabee would you expect all transactions in the current "Session" or from a store that lasts longer?

Reply:
I'm ambivalent. 

Doing this for the current session is obviously the simplest option, but if the data is local and encrypted with the wallet password then there's no reason it couldn't be a longer history. 

If it is longer it _should_ be possible to delete individual items and to clear the whole history, and I _would like_ to be able to set a maximum age / expiry time for history items after which they are deleted.

Reply:
okay I'm going to make this one Session specific.
I'm going to add the requirements barney mentioned now (so not lost to this thread), but comment them out until we agree they are in scope.

Reply:
resolved in: https://github.com/vegaprotocol/specs/pull/1181/commits/94917202298a2d9393ec04659dfd55de8b4304c1

Reply:
have also raise a ticket for the wallet backlog to implement long term storage when appropriate https://github.com/vegaprotocol/vega/issues/6142